Makefile

```diff
@@ -1,47 +1,21 @@
-OPTCC = /opt/at11.0/bin/gcc
-ifneq ("$(wildcard $(OPTCC))","")
-	CC = $(OPTCC)
-else
-	CC = gcc
-endif
-FLG = -std=gnu11
-SFLAGS = -O3 -fPIC -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN
-ZLIB = -DZLIB_API
-CFLAGS = $(FLG) $(SFLAGS) $(ZLIB) -mcpu=power9 #-DNXTIMER
 
-SRCS = nx_inflate.c nx_deflate.c nx_zlib.c nx_crc.c nx_dht.c nx_dhtgen.c nx_dht_builtin.c \
-       nx_adler32.c gzip_vas.c nx_compress.c nx_uncompr.c crc32_ppc.c crc32_ppc_asm.S nx_utils.c
-OBJS = nx_inflate.o nx_deflate.o nx_zlib.o nx_crc.o nx_dht.o nx_dhtgen.o nx_dht_builtin.o \
-       nx_adler32.o gzip_vas.o nx_compress.o nx_uncompr.o crc32_ppc.o crc32_ppc_asm.o nx_utils.o
+subdirs = lib test samples
 
-VERSION ?= $(shell git describe --tags | cut -d - -f 1,2 | tr - . | cut -c 2-)
-SOVERSION = $(shell echo $(VERSION) | cut -d . -f 1)
-PACKAGENAME = libnxz
-STATICLIB = $(PACKAGENAME).a
-SHAREDSONAMELIB = $(PACKAGENAME).so.$(SOVERSION)
-SHAREDLIB = $(PACKAGENAME).so.$(VERSION)
-LIBLINK = $(PACKAGENAME).so
+all: $(subdirs)
 
-INC = ./inc_nx
+.PHONY: $(subdirs)
+$(subdirs):
+	@if [ -d $@ ]; then				\
+		$(MAKE) -C $@ || exit 1;	 	\
```

Comment:
I don't think it's a good practice to build the samples and the tests with the library, users usually only want to build the library and will have to wait until the rest is built.

Reply:
If users want to only build the library, they can cd into the library directory and make. Build the lib, samples, test together can avoid incompatible between lib and sample/test.

Reply:
@davidzengxhsh Yes, but the root directory is the go to place to build many libraries, so by seeing a Makefile there most people will just try make, and our README implies that you should do that.
The tests are interesting for the developers and the C code on samples is usually enough to understand the usage. I still believe that is better to just build the library in this Makefile, but I'll not block the commit.